core/src/main/scala/org/zalando/grafter/Rewriter.scala

```diff
@@ -173,6 +175,22 @@ trait Rewriter {
 
     results.toList
   }
+
+  def topBreadthfirst(s: Strategy): Strategy = {
+    def bf(s1: =>Strategy): Strategy =
+      one(s1) <+ one(bf(s1))
```

Comment:
That's a strategy combinator: `a <+ b` means try `a`, if it doesn't succeed, try `b`.